src/Umbraco.Core/Persistence/Mappers/BaseMapper.cs

```diff
@@ -10,22 +8,38 @@ namespace Umbraco.Core.Persistence.Mappers
 {
     public abstract class BaseMapper : IDiscoverable
     {
-        protected BaseMapper()
+        private readonly ISqlSyntaxProvider _sqlSyntax;
+        private readonly ConcurrentDictionary<Type, ConcurrentDictionary<string, string>> _maps;
+
+        protected BaseMapper(ISqlContext sqlContext, ConcurrentDictionary<Type, ConcurrentDictionary<string, string>> maps)
         {
-            Build();
+            _sqlSyntax = sqlContext.SqlSyntax;
+            _maps = maps;
         }
 
-        internal abstract ConcurrentDictionary<string, DtoMapModel> PropertyInfoCache { get; }
-
-        private void Build()
+        internal string Map(string propertyName, bool throws = false)
         {
-            BuildMap();
+            if (!_maps.TryGetValue(GetType(), out var mapperMaps))
+                throw new InvalidOperationException($"No maps defined for mapper {GetType().FullName}.");
+            if (!mapperMaps.TryGetValue(propertyName, out var mappedName))
+                throw new InvalidOperationException($"No map defined by mapper {GetType().FullName} for property {propertyName}.");
+            return mappedName;
         }
 
-        protected abstract void BuildMap();
-
+        // fixme kill
```

Comment:
Should you kill this then?

Reply:
I've been lazy and forgot to check for fixmes before pushing - clear now.